internal/cmd/server_test.go

```diff
@@ -15,3 +16,19 @@ func TestGetServerCommand(t *testing.T) {
 	got := getServerCommand()
 	assert.Equal(t, "server", got.Use)
 }
+
+func TestValidateUserConfig(t *testing.T) {
```

Comment:
would be great if the test looks like https://github.com/envoyproxy/gateway/blob/3c69c36f6de255d0532cb6a1ca245a4494ff4124/internal/ir/xds_test.go#L570 so its easier to add more test cases in the future